### PR TITLE
create data model

### PR DESCRIPTION
**Pull request by Tobby8629**
- create ruby on rails app connected to postgres database
- create models for each table to manage data
- tables includes users, comments, posts and likes
- create the foreign keys to the tables
- migrate the data for schema update

### CODE REVIEW

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @Tobby8629, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't use professional readme template. Please use [this](https://github.com/microverseinc/readme-template)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/schema.rb [hunk 1]
```diff
@@ -10,8 +10,50 @@
...
+
+  create_table "likes", force: :cascade do |t|
+    t.bigint "post_id_id", null: false
+    t.bigint "author_id_id", null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["author_id_id"], name: "index_likes_on_author_id_id"
+    t.index ["post_id_id"], name: "index_likes_on_post_id_id"
+  end
+
```
Comment (TedLivist) [inline db/schema.rb]: - [ ] Kindly note that these should be `post_id` and `author_id`, not `post_id_id` and `author_id_id`. Kindly fix the errors.

File: db/schema.rb [hunk 2]
```diff
@@ -10,8 +10,50 @@
...
+  create_table "posts", force: :cascade do |t|
+    t.bigint "author_id_id", null: false
+    t.string "title"
+    t.text "text"
+    t.integer "commentCounter"
+    t.integer "likeCounter"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["author_id_id"], name: "index_posts_on_author_id_id"
+  end
```
Comment (TedLivist) [inline db/schema.rb]: - [ ] Kindly note that these should be `author_id`, not `author_id_id`.

- [ ] Let's stick to the naming conventions in Ruby. We should use underscore for separating table column name, not camel case. Kindly rename the following table columns `commentCounter` and `likeCounter` to `comments_counter` and `likes_counter` respectively.

File: db/schema.rb [hunk 3]
```diff
@@ -10,8 +10,50 @@
...
+
+  create_table "users", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.string "name"
+    t.text "photo"
+    t.string "bio"
+    t.integer "postsCounter"
```
Comment (TedLivist) [inline db/schema.rb]: - [ ] Let's stick to the naming conventions in Ruby. We should use an underscore for separating table column names, not camel cases. Kindly rename the following table columns `postsCounter` to `posts_counter`.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Tobby8629,

Good job so far!
✔️ PostgreSQL is used for database
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @Tobby8629 , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Tobby8629** [comment]
@TedLivist @sumairq  Thank you for the good code review 🙏🏾🙏🏾

**Pull request merged**